File: dataprovider-retrofit/src/main/java/ch/srg/dataProvider/integrationlayer/SRGUrlFactory.java
package ch.srg.dataProvider.integrationlayer;

import android.net.Uri;

import ch.srg.dataProvider.integrationlayer.request.IlHost;

/**
 * Copyright (c) SRG SSR. All rights reserved.
 * <p>
 * License information is available from the LICENSE file.
 */
public class SRGUrlFactory {
    private final Uri hostUri;
    private final Uri ilBaseUri;

    public SRGUrlFactory(IlHost ilHost) {
        hostUri = ilHost.getHostUri();
        ilBaseUri = Uri.parse(ilHost.getHostUri() + "/integrationlayer/");
    }

    public Uri getHostUri() {
        return hostUri;
    }

    public Uri getIlBaseUrl() {
        return ilBaseUri;
    }

}

File: dataprovider-retrofit/src/main/java/ch/srg/dataProvider/integrationlayer/SRGUrlFactory.kt
package ch.srg.dataProvider.integrationlayer

import android.net.Uri
import ch.srg.dataProvider.integrationlayer.request.IlHost

/**
 * SRG Url factory
 *
 * @param ilHost IlHost to create service urls.
 */
class SRGUrlFactory(ilHost: IlHost) {
    val hostUri: Uri = ilHost.hostUri
    val ilBaseUrl: Uri = Uri.parse("${ilHost.hostUri}/integrationlayer/")

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as SRGUrlFactory

        if (hostUri != other.hostUri) return false
        if (ilBaseUrl != other.ilBaseUrl) return false

        return true
    }

    override fun hashCode(): Int {
        var result = hostUri.hashCode()
        result = 31 * result + ilBaseUrl.hashCode()
        return result
    }

    override fun toString(): String {
        return "SRGUrlFactory(hostUri=$hostUri, ilBaseUrl=$ilBaseUrl)"
    }
}
